src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/Tree.java

```diff
@@ -252,17 +268,32 @@ public static boolean checkTree(@NotNull final IBlockAccess world, final BlockPo
         return new Tuple<>(bottom, top);
     }
 
-    private static boolean hasEnoughLeaves(@NotNull final IBlockAccess world, final BlockPos pos)
+    /**
+     * Check if the tree has enough leaves and the lj is supposed to cut them.
+     * @param world the world it is in.
+     * @param pos the position.
+     * @param treesToCut the trees the lj is supposed to cut.
+     * @return true if so.
+     */
+    private static boolean hasEnoughLeavesAndIsSupposedToCut(@NotNull final IBlockAccess world, final BlockPos pos, final Map<ItemStorage, Boolean> treesToCut)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)